W-10588843-file-copy-kt (#2071)
mulesoft/docs-connectors: file/1.5/modules/ROOT/pages/file-copy-move.adoc

= Copy and Move Files - Mule 4
:page-aliases: connectors::file/file-copy-move.adoc

Anypoint Connector for File (File Connector) provides the *Copy* and *Move* operations that enable you to copy or move files or directories on demand. 

* *Source path* +
The directory path name or file name to copy or move. If a file or directory already exists and the *Overwrite* field is set to `false`, the connector throws the message `FILE:FILE_ALREADY_EXISTS`.

* *Target path* +

>  The directory path name into which to copy or move the file. If a target path doesn't exist, and neither does its parent, the connector makes an attempt to create the path depending on the value of the *Create parent directories* argument. If *Create parent directories* is `false`, then `FILE:ILLEGAL_PATH` occurs.

Reviewer: ```suggestion
 The directory path name into which to copy or move the file. If a target path doesn't exist, and neither does its parent, the connector attempts to create the path depending on the value of the *Create parent directories* argument. If *Create parent directories* is `false`, then `FILE:ILLEGAL_PATH` occurs.
```

* *Rename to* +

> Renames the file as part of the operation. Set this field with file name, not a path. The connector retains the original file name, if you do not configure this field.

Reviewer: ```suggestion
Renames the file as part of the operation. Set this field with the file name and not a path. The connector retains the original file name if you do not configure this field.
```

> Renames the file as part of the operation. Set this field with file name, not a path. The connector retains the original file name, if you do not configure this field.

Reviewer: Suggestion here and wherever else applicable:
Set this field with a file name instead of a path.

> Renames the file as part of the operation. Set this field with file name, not a path. The connector retains the original file name, if you do not configure this field.

Reviewer: ```suggestion
Renames the file as part of the operation. Set this field with file name, not a path. The connector retains the original file name if you do not configure this field.
```
